src/nvim/runtime.c

```diff
@@ -847,21 +835,69 @@ int source_in_path(char *path, char *name, int flags)
   return do_in_path_and_pp(path, name, flags, source_callback, NULL);
 }
 
-// Expand wildcards in "pat" and invoke do_source()/nlua_exec_file()
-// for each match.
-static void source_all_matches(char *pat)
+static bool str_ends_with(const char *str, const char *suffix)
+{
+  if (!str || !suffix) {
+    return false;
+  }
+  size_t lenstr = strlen(str);
+  size_t lensuffix = strlen(suffix);
+  if (lensuffix > lenstr) {
+    return false;
+  }
+  return strncmp(str + lenstr - lensuffix, suffix, lensuffix) == 0;
+}
+
+/// Expand wildcards in "pat" and invoke do_source()/nlua_exec_file()
+/// for each match.
+/// @param pat pattern to expand
+static bool expand_wildcards_and_cb(char *pat, int ew_flags, bool all, DoInRuntimepathCB callback, void *cookie)
```

Comment:
perhaps `expand_wildcards_and_source` and ? This usecase seems to be the implied one given that it hardcodes matching of .vim then .lua files

Reply:
This is also called with `add_pack_start_dir` and `add_start_pack_plugin` which in those cases the first two loops (for `.lua` and `.vim`) don't do anything.

Reply:
Modifying the signature of `DoInRuntimepathCB` to support multiple files seems to be a better route.